### ISSUE TITLE
Errors that occur at the very end of the line somehow extend the line to the start of the next one

### ISSUE THREAD

**Issue by johnhamelink** [konrad1977/flycheck-overlay#32]
In this example, I have written a sentence and added a space, then moved away from that line or paused for a duration longer than the debounce value. This triggers a flymake diagnostic due to there being a trailing space on that line - that all makes sense. However, something about how flyover renders the diagnostic results in the end of line character position descending (visually) into the next line, as if there was a word wrap on that line. When flyover disappears, the issue also resolves.

Before fixing the diagnostic: 
<img width="688" height="566" alt="Image" src="https://github.com/user-attachments/assets/75368a02-bfbb-4b40-bc4b-f8fc67b43c4d" />

After fixing the diagnostic:
<img width="707" height="567" alt="Image" src="https://github.com/user-attachments/assets/169e3527-0efd-43cf-8373-ffceb386aecf" />

My flyover configuration:

```elisp
(use-package flyover
  :ensure t
  :hook ((flycheck-mode . flyover-mode)
         (flymake-mode . flyover-mode))
  :custom
  ;; Checker settings
  (flyover-checkers '(flycheck flymake))
  (flyover-levels '(error warning info))

  ;; Appearance
  (flyover-use-theme-colors t)
  (flyover-background-lightness 45)
  (flyover-percent-darker 40)
  (flyover-text-tint 'lighter)
  (flyover-text-tint-percent 50)

  ;; Icons
  (flyover-info-icon "ℹ")
  (flyover-warning-icon "⚠")
  (flyover-error-icon "✘")

  ;; Display settings
  (flyover-hide-checker-name t)
  (flyover-show-virtual-line nil)
  (flyover-virtual-line-type 'line-no-arrow)
  (flyover-line-position-offset 1)
  (flyover-show-error-id t)

  ;; Message wrapping
  (flyover-wrap-messages t)
  (flyover-max-line-length 80)

  ;; Performance
  (flyover-debounce-interval 0.2))
```

**Comment by xendk** [comment]
This is very confusing. Take this YAML example:

<img width="673" height="95" alt="Image" src="https://github.com/user-attachments/assets/d11ff63a-22da-448c-b24c-c3d5c58994c2" />

One's instinct is to move the cursor, but moving up ends up on the environment line, and going left, you're over the final `b`. If you're coming from below and the cursor is at a horizontal position greater than the line with the error, it moves to the start of the error line.

It totally messes with ones musclememory, resulting in a longer dance trying to get the cursor to the position that your brain expects (right after the final `b`), but not being able to get there. Until one remembers this issue and start writing on the wrong line.

It would be very nice to get fixed.
